### PR TITLE
[REVIEW] Fix cudf legacy Cython imports/cimports

### PR DESCRIPTION
**Pull request by shwina**
This PR fixes `cudf` Cython imports/cimports following https://github.com/rapidsai/cudf/pull/4627. This PR should be merged afterr https://github.com/rapidsai/cudf/pull/4627 is merged.

### CODE REVIEW

File: python/cuspatial/cuspatial/_lib/soa_readers.pxd [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # cython: language_level = 3
```
Comment (kkraus14) [inline python/cuspatial/cuspatial/_lib/soa_readers.pxd]: Copyright in this file

File: python/cuspatial/cuspatial/_lib/shapefile_reader.pxd [hunk 2]
```diff
@@ -4,7 +4,7 @@
 # cython: language_level = 3
```
Comment (kkraus14) [inline python/cuspatial/cuspatial/_lib/shapefile_reader.pxd]: Copyright in this file

**Comment by kkraus14** [comment]
rerun tests

**Comment by mike-wendt** [comment]
rerun tests

**Pull request merged**